### PR TITLE
Fix copy title from metadata

### LINKED ISSUE #2516: "Copying Title" function stopped working for create post
### Requirements

- [X] This is a bug report, and if not, please post to https://lemmy.ml/c/lemmy_support instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this issue already exists.
- [X] It's a single bug. Do not report multiple bugs in one issue.
- [X] It's a frontend issue, not a backend issue; Otherwise please create an issue on the [backend repo](https://github.com/LemmyNet/lemmy) instead.

### Summary

The title copying button no longer works.

Likely this PR broke title copying: https://github.com/LemmyNet/lemmy-ui/pull/2515

### Steps to Reproduce

1. Go to the create post page
2. Type in a url
3. See the metadata fetched title
4. Click copy
5. Nothing happens

### Technical Details

NA

### Lemmy Instance Version

0.19.4

### Lemmy Instance URL

https://voyager.lemmy.ml

### PR DESCRIPTION
**Pull request by SleeplessOne1917**
Fixes #2516 

Alternative to #2517

### CODE REVIEW

**Comment by dessalines** [comment]
Hrm... didn't work. 

Click it the first time, it doesn't copy. Alter the URL again, then click copy, and it works the 2nd time only.

**Comment by SleeplessOne1917** [comment]
I see the issue. It works the first time if the url field loses focus, but it doesn't work the first time when you go from the url field directly to the copy button.

File: src/shared/components/post/post-form.tsx [hunk 1]
```diff
@@ -175,17 +191,19 @@ function handlePostUrlChange(i: PostForm, event: any) {
...
 
 function handlePostUrlBlur(i: PostForm, event: any) {
-  i.setState({ bypassNavWarning: true });
-  i.props.onUrlBlur?.(event.target.value);
-  i.setState({ bypassNavWarning: false });
+  i.setState({
+    urlBlurTimeout: setTimeout(() => {
+      i.updateUrl(() => i.props.onUrlBlur?.(event.target.value));
+    }, 500),
+  });
```
Comment (SleeplessOne1917) [inline src/shared/components/post/post-form.tsx:198]: A kludge. It fixes the bug at least.

**Review by dessalines — APPROVED** [review]
It still does some reloading, but the filling works correctly now. Good enough for me, and we can fix the other things in the next release.

File: src/shared/components/post/post-form.tsx [hunk 2]
```diff
@@ -354,6 +367,10 @@ export class PostForm extends Component<PostFormProps, PostFormState> {
         },
       };
     }
+
+    if (this.state.form.url) {
+      this.fetchPageTitle();
```
Comment (matc-pub) [inline src/shared/components/post/post-form.tsx:372]: This calls setState in the constructor. Probably only a problem for dev builds.

Comment (matc-pub) [reply src/shared/components/post/post-form.tsx:372]: And should also be client side only.

**Comment by dessalines** [comment]
There are still a few issues with both cross-posting, and copy title. We can fix them for the next bugfix release tho.

**Pull request merged**